File: Code Examples/Heavy/HeavyCorrectExample.java
class Solution {
    public double fractionalKnapsack(Item[] items, int capacity) {
    java.util.Arrays.sort(items, (a, b) -> Double.compare(b.getRatio(), a.getRatio()));

    double totalValue = 0.0;

    for (Item item : items) {
        if (capacity >= item.weight) {
            capacity -= item.weight;
            totalValue += item.value;
        } else {
            totalValue += item.getRatio() * capacity;
            break;
        }
    }

    return totalValue;
}

}

File: Code Examples/Heavy/HeavyCorrectExample.swift
class Solution {
    func fractionalKnapsack(_ items: [Item], _ capacity: Int) -> Double {
        let sortedItems = items.sorted { $0.ratio > $1.ratio }
        var remainingCapacity = capacity
        var totalValue: Double = 0.0

        for item in sortedItems {
            if remainingCapacity >= item.weight {
                remainingCapacity -= item.weight
                totalValue += Double(item.value)
            } else {
                totalValue += item.ratio * Double(remainingCapacity)
                break
            }
        }

        return totalValue
    }
}